### DISCUSSION TITLE
Export Workflow History

### DISCUSSION THREAD

**Discussion by markadrake** [nathanwoulfe/Plumber-2#141] [Ideas]
My client has requested a feature. For auditing purposes, they'd like to export the workflow history. I didn't receive any parameters (CSV, XLSX, JSON) but I do know they wish to report this information to an external authority (someone without access to Umbraco). 

Would other people find an export beneficial?

> I'd gladly contribute to the project Nathan if the opportunity presents itself. (*Umbraco Gold partner).

![image](https://user-images.githubusercontent.com/713355/196997306-8969ee97-0e24-417e-adf0-80a310c0144c.png)

**Comment by nathanwoulfe** [comment]
Hey @markadrake 

I have no problem adding this as a new feature - would be straightforward to return JSON, probably equally straightforward to return XML, but CSV would be a bit of a mess given the nested/enumerable properties belonging to each workflow. Could export the workflows and their tasks as two separate CSVs, which would then need some Excel magic to link the two sheets (speculating, I'd imagine that would be possible).

If you can find out more detail around formats, I can pull this into the backlog
